### PR TITLE
autogen.pl: aborts if autogen.pl is invoked from an Open MPI tarball …

### PR DESCRIPTION
**Pull request by ggouaillardet**
…and without the --force option

### CODE REVIEW

File: autogen.pl [hunk 1]
```diff
@@ -1020,6 +1021,27 @@ sub patch_autotools_output {
...
+sub in_tarball {
+    my $tarball = 0;
+    open(IN, "VERSION") || my_die "Can't open VERSION";
+    # Read all lines from the file
+    while (<IN>) {
+          my $line = $_;
+          my @fields = split(/=/,$line);
+          if ($fields[0] eq "repo_rev") {
+              if ($fields[1] ne "\n") {
+                  $tarball = 1;
```
Comment (jsquyres) [inline autogen.pl]: Minor optimization: add a `last;` here to break out of the loop.

File: autogen.pl [hunk 2]
```diff
@@ -1020,6 +1021,27 @@ sub patch_autotools_output {
...
+              }
+          } elsif ($fields[0] eq "tarball_version") {
+              if ($fields[1] ne "gitclone\n") {
```
Comment (jsquyres) [inline autogen.pl]: Are you checking for `tarball_version=gitclone` simply as defensive programming?  (I.e., it should always catch the `repo_rev=` check, but might as well put in the check for `tarball_version`, too?)

My $0.02: Might be worth putting in a comment here explaining why you're checking `tarball_version`.

File: autogen.pl [hunk 3]
```diff
@@ -1056,7 +1079,9 @@ sub patch_autotools_output {
                                 will be ignored and only those specified will be marked
                                 to build
   --exclude | -e                Comma-separated list of framework or framework-component
-                                to be excluded from the build\n";
+                                to be excluded from the build
+  --force | -f                  Run autogen.pl even if invoked from a source tree coming
+                                from a tarball\n";
```
Comment (jsquyres) [inline autogen.pl]: Suggestion:

> Run even if invoked from the source tree of an expanded distribution tarball.

File: autogen.pl [hunk 4]
```diff
@@ -1120,6 +1145,11 @@ sub patch_autotools_output {
 my_die "Not at the root directory of an OMPI source tree"
     if (! -f "config/opal_try_assemble.m4");
 
+my_die "autogen.pl has been invoked in a source tree coming from an OMPI tarball, aborting...
+you should instead directly invoke configure
+if you really want to run autogen.pl, and you know what you are doing, run autogen.pl with the --force flag"
+    if (!$force_arg && in_tarball());
```
Comment (jsquyres) [inline autogen.pl]: Suggestion:

> autogen.pl has been invoked in the source tree of an Open MPI distribution tarball; aborting...
> You likely do not need to invoke "autogen.pl" -- you can probably run "configure" directly.
> If you *really* know what you are doing, and *really* need to run autogen.pl, use the "--force" flag.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by ggouaillardet** [comment]
:bot:retest

**Comment by jsquyres** [comment]
:+1:

**Comment by jsquyres** [comment]
@ggouaillardet You can probably squash these down to 1 commit, but otherwise it looks good.

**Pull request merged**